### ISSUE TITLE
Change the Discord Bot from Denizen to PHP

### ISSUE THREAD

**Issue by nomadjimbob** [Drustcraft/engine#91]
### What feature do you want to see added?

Currently the discord bot uses Denizen to work closely with the Minecraft server.

The issue is that if the server is offline, the Discord bot is also offline.

It would be better if it was a PHP script as it would be online  and available while the server is offline and under maintenance. It would also allow integration with GitHub with issues and tickets.

**Comment by ProjectHSI** [comment]
oh boy i gotta learn discordphp now

**Comment by ProjectHSI** [comment]
not that i already have to use the documentation and existing possibly bad code from /// HSI Protector System for discord.js

**Comment by ProjectHSI** [comment]
PHEW
Is Discord.JS fine to be used? i just don't wanna learn an entirely seperate language.

**Comment by nomadjimbob** [comment]
The bot does not need to be in PHP. As long as it can be run from a ubuntu server that is fine. We would need to work out how the bot would communicate with the Minecraft server and vise versa using the Drustcraft API (more endpoints would probably be needed)

**Comment by ProjectHSI** [comment]
Discussed in Discord
Final Addition: Discord Linking

**Comment by ProjectHSI** [comment]
ares slash commands required?

**Comment by nomadjimbob** [comment]
Yes. It is best that future bots use Discord Slash Command so that users can see how to use a command as well as having validation and error checking built-in

**Comment by ProjectHSI** [comment]
Alright, this is because discordphp (the library I was using) doesn't have documentation for slash commands.

**Comment by nomadjimbob** [comment]
No, the docs are missing around that. However you can gather some info here - https://github.com/discord-php/DiscordPHP/blob/8e527b443b726971e90fe61f43d10ab1bef1e93c/V7_CONVERSION.md and searching for `Create a Global Command`.

Then you can include an array of 'options' that the slash command supports using the class at https://github.com/discord-php/DiscordPHP/blob/9601ea9a86e3c9234459c8f52c4cce765c57152c/src/Discord/Parts/Interactions/Command/Option.php

```
$options[] = new Option()->setType(STRING)->setName('Code')->setDescription('Linking code from Minecraft')->setRequired(true);

$command = new Command($discord, ['name' => 'link', 'description' => 'Enter a linking code from Minecraft', $options);
$discord->application->commands->save($command)
```

That should create a /link command that requires a string based code. All the user help should also be displayed within Discord.

I then think you can then process the command by adding this before the event loop (not 100%)

```
$discord->listenCommand('link', function (Interaction $interaction) {
  $interaction->acknowledge(true);
  if($interaction->data->options['code'] != '1111') {
    $interaction->replyWithSource('Thats right!');
  } else {
    $interaction->replyWithSource('Incorrect code');
  }
});
```
